src/main/java/PIDControl/PIDControl.java

```diff
@@ -270,14 +282,10 @@ public void setSetpoint(double setpoint, double inputNow, boolean resetPID)
         }
 
         _setpoint = setpoint;
-        _errorPrev = _setpoint - inputNow;
-        _errorIsPositive = _errorPrev >= 0.0;
```

Comment:
If we remove setting `_errorPrev` at this point, our `D` coefficients won't behave properly for one timestep when we call the `calculate` method.

Say our PID setpoint and current position are at a value of 100, then we change our setpoint to 0.

In the old implementation, `_errorPrev` gets set to `-100`. Then, when we call `calculate`, our `error` is calculated as `-100` as well, meaning our `errorDiff` (used with the `D` coefficient) is `0`.

In this implementation, `_errorPrev` isn't changed (and is presumably pretty close to `0`). Then, when we call `calculate`, our error is still calculated as `-100`, meaning our `errorDiff` is `-100`.

This is a little bit of an exaggerated example, but it's possible that with an errorDiff that large, we could see some behavior where the system "jerks back" in the wrong direction (to counteract the incorrect diff measurement) for a frame before kicking in correctly in future frames.